src/java/us/kbase/mobu/ModuleBuilder.java

```diff
@@ -360,24 +361,32 @@ private static int runTestCommand(TestCommandArgs testArgs, JCommander jc) {
     }
 
     private static void printVersion() {
-        String gitCommit = getGitCommit();
-        System.out.println("KBase SDK version " + VERSION + (gitCommit == null ? "" : (" (commit " + gitCommit + ")")));
+        String gitCommit = getGitProp("commit");
+        String timestamp = getGitProp("build_timestamp");
+        System.out.println("KBase SDK version " + VERSION
+            + (timestamp == null ? "" : ("_" + timestamp))
+            + (gitCommit == null ? "" : (" (commit " + gitCommit + ")")));
     }
 
-    public static String getGitCommit() {
-        String gitCommit = null;
+
+    public static String getGitProp(String propertyToGet) {
+        String propertyValue = null;
+        Properties gitProps = new Properties();
         try {
-            Properties gitProps = new Properties();
             InputStream is = ModuleBuilder.class.getResourceAsStream("git.properties");
             gitProps.load(is);
             is.close();
-            gitCommit = gitProps.getProperty("commit");
-        } catch (Exception ignore) {}
-        return gitCommit;
+            propertyValue = gitProps.getProperty(propertyToGet);
+        } catch (Exception e) {
+            showError("Error while retrieving version information from git.properties file: ", e.getMessage());
+            System.exit(1);
```

Comment:
When I was looking at the module builder code earlier, I noticed that most of the methods returned the return code rather than calling exit() themselves. Does that need to happen here? I don't actually understand why they were passing back the return code, but I assume it's for some good reason.

Reply:
Maybe testability? If you return the code, you can test the method, but not if you call exit().

Reply:
This happens before any of the module builder methods are called, so you have to just exit. The module builder methods all expect an integer (representing success or otherwise) to be returned:

```java
// code adding the various valid commands and args to JCommander

// ModuleBuilder line 97: print name and version
printVersion();

// parse the arguments and gracefully catch any errors
try {
    jc.parse(args);
} catch (RuntimeException e) {
    showError("Command Line Argument Error", e.getMessage());
    System.exit(1);
}

// snip some help command stuff

// if we get here, we have a valid command, so process it and do stuff
int returnCode = 0;
if(jc.getParsedCommand().equals(HELP_COMMAND)) {
    showCommandUsage(jc,help,System.out);
} else if(jc.getParsedCommand().equals(INIT_COMMAND)) {
    returnCode = runInitCommand(initArgs,jc);
// etc.
```